lib/proxy/peer/quicserver.go

```diff
@@ -78,31 +91,335 @@ func (c *QUICServerConfig) checkAndSetDefaults() error {
 }
 
 // QUICServer is a proxy peering server that uses the QUIC protocol.
-type QUICServer struct{}
+type QUICServer struct {
+	log           *slog.Logger
+	clusterDialer ClusterDialer
+	tlsConfig     *tls.Config
+	quicConfig    *quic.Config
+
+	mu     sync.Mutex
+	closed bool
+	wg     sync.WaitGroup
+
+	replayStore replayStore
+
+	// runCtx is a context that gets canceled when all connections should be
+	// ungracefully terminated.
+	runCtx context.Context
```

Comment:
Do you really need a context or is this a glorified channel? Bear in mind the usual advice of not having context fields.

Same for serveCtx.

- https://go.dev/wiki/CodeReviewComments#contexts

Reply:
We make use of `context.AfterFunc` for both.